feat(hovercard): DLT-2845 add focus trap to hovercard content (#1330)
dialpad/dialtone: apps/dialtone-documentation/docs/components/hovercard.md

<!-- <component-combinator component-name="DtHovercard" /> -->

## Usage

> - Hovercard is a **progressive enhancement**. It surfaces supplementary information about an entity or provides convenience shortcuts on hover, without navigating away from the current view. Every action it exposes **must also be reachable by another route** (a dedicated page, a menu, or an inline button).

Reviewer: ```suggestion
- Hovercard is a **progressive enhancement**. It surfaces supplementary information about an entity or provides convenience shortcuts on hover, without navigating away from the current view. Its content or functionality **must also be reachable by another route** (a dedicated page, a menu, or an inline button).
```

> - Keep content concise. Hovercards are transient overlays — deep interactions, multi-step flows, or extensive content belong in a dedicated UI, or in a [Modal](modal.md).

Reviewer: ```suggestion
- Keep content concise. Hovercards are transient overlays, and are not appropriate for deep interactions, multi-step flows, or extensive content.
```

- Use Hovercard for contextual content with structure or actions. Use [Tooltip](tooltip.md) for brief, text-only descriptions of a control. Use [Popover](popover.md) when the content is triggered by a deliberate click and should persist until dismissed.
- `enterDelay` is used to prevent the card from triggering on accidental cursor passes. The default delay of 300 ms is recommended for most contexts.

<dialtone-usage>
<template #do>

- Surface contextual details like a user's status, avatar, and quick-action shortcuts.
- Treat hovercard actions as shortcuts — ensure the same actions are reachable elsewhere in the UI.

</template>
<template #dont>

- Use Hovercard as the only way to reach content or perform an action.
- Place critical or destructive actions exclusively inside a hovercard.
- Use for deep interactions or lengthy content — prefer a dedicated UI or a [Modal](modal.md).

</template>
</dialtone-usage>

## Accessibility

> When focus moves into an open hovercard, focus is trapped within it — the user can Tab between focusable elements inside the card without accidentally leaving. Pressing Escape dismisses the card and restores focus to the element that had focus before the card opened.

Reviewer: ```suggestion
When focus moves into an open hovercard, focus is trapped within. The user can Tab between focusable elements inside the card without accidentally leaving. Clicking outside the hovercard or keypress of `esc` will dismiss the card and restore focus to the element that had focus before the card opened.
```

> When focus moves into an open hovercard, focus is trapped within it — the user can Tab between focusable elements inside the card without accidentally leaving. Pressing Escape dismisses the card and restores focus to the element that had focus before the card opened.

Reviewer: > "Pressing Escape dismisses the card and restores focus to the element that had focus before the card opened."

This is currently not true. In attached video, I...

1. Focus in middle of the page.
2. Hover over top nav, tab within trapped Hovercard.
3. Hit escape...
  1. At this point, I expect the focus to go to the last item I had focused before bringing up hovercard.
  2. It currently appears to go to the first focusable element at root.

Should we keep track of the last focused item once the Hovercard has focus within? 

https://github.com/user-attachments/assets/090980d7-4c61-4963-aa6e-0d341af66133

Reviewer: Beacon example.

https://github.com/user-attachments/assets/8218e569-a05f-4f7c-a7f5-521b0021c3a0

Author: Hmmm, this should've been fixed with https://github.com/dialpad/dialtone/pull/1330/changes/ddc769934e9fec48adde7887d629e49ebd3e463a 

https://github.com/user-attachments/assets/95ac7f84-428c-49c7-b517-ef59a3dd986c

Try again please and let me know if you're still seeing the issue

Reviewer: Confirmed. It does. Merge away!

https://github.com/user-attachments/assets/66f42de1-d0f8-46e4-8078-838bd318f82a
